### ISSUE TITLE
[LintRule] Make color token definition sites configurable for avoid_hardcoded_color

### ISSUE THREAD

**Issue by Ituking** [altive/altive_lints#133]
## Summary

`avoid_hardcoded_color` currently treats `ColorScheme` as the only place where a color may be defined. Projects that centralize their color tokens in another form — a class of `static const Color` fields, or a `ThemeExtension` — have no way to exempt their token definition file, so the file that is *supposed* to hold the raw color values is reported as one violation per token.

This is the same concern as #67, generalized: the principle "do not warn at the place where colors are defined" is already accepted, but its implementation is tied to one specific definition style.

## Background

Our project routes every color through a single token definition and forbids direct color literals everywhere else. The rule this enforces exists because a hardcoded color in a widget silently breaks theming: a literal dark text color stays dark in dark mode and becomes unreadable, and nothing fails at build time or in tests — it only shows up when someone happens to look at the screen in the other theme. Static detection is exactly the right tool for this, which is why `avoid_hardcoded_color` is valuable to us.

The obstacle is not the detection itself but the exemption. Our token definition is a plain class:

```dart
class AppColors {
  const AppColors._();

  static const Color primary = Color(0xFF1A73E8);
  static const Color surface = Color(0xFFF5F5F5);
  // ...dozens more
}
```

Every one of these lines is reported, because `_isInsideColorScheme` only walks up for an `InstanceCreationExpression` (or method call) whose static type is `ColorScheme`. A `static const Color` field in an ordinary class never matches.

The only available workaround is a file-level suppression:

```dart
// ignore_for_file: altive_lints_plugin/avoid_hardcoded_color
```

That works, but it expresses "this file is the single source of truth for colors" as a blanket suppression comment rather than as configuration, and it also silences any genuinely accidental literal that later gets added to that file.

## Why this is not specific to one project

Defining color tokens as `static const Color` fields, or via `ThemeExtension`, is a common approach in Flutter — `ThemeExtension` is a first-party mechanism for exactly this purpose. A rule that recognizes only `ColorScheme` as a definition site fits one of several mainstream styles.

Adding a dedicated exclusion branch per style (`ThemeExtension`, then constant classes, then whatever comes next) does not scale and keeps requiring a release of the plugin for each new case. Letting the user declare where their tokens live solves all of them at once.

## What should be detected (unchanged)

No change is proposed to detection. These should keep being reported:

```dart
ColoredBox(color: Color(0xFF00FF00));          // LINT
ColoredBox(color: Color.fromARGB(255, 0, 0, 0)); // LINT
ColoredBox(color: Color.fromRGBO(0, 0, 0, 1));   // LINT
ColoredBox(color: Colors.red);                   // LINT
```

## What should not be detected

Already handled today:

- `Colors.transparent` (#59)
- inside a `ColorScheme(...)` definition (#67)
- test files (#82)
- explicit `// ignore:` / `// ignore_for_file:` comments

Not handled today, and the subject of this issue:

```dart
// lib/theme/app_colors.dart — the project's single source of truth
class AppColors {
  static const Color primary = Color(0xFF1A73E8); // LINT (should be allowed)
}

// lib/theme/brand_colors.dart — ThemeExtension variant
class BrandColors extends ThemeExtension<BrandColors> {
  const BrandColors({required this.accent});
  final Color accent;

  static const light = BrandColors(accent: Color(0xFF1A73E8)); // LINT (should be allowed)
}
```

## Proposed configuration

Allow the rule to take a list of paths that are exempt:

```yaml
plugins:
  altive_lints_plugin:
    version: ^1.0.0
    diagnostics:
      avoid_hardcoded_color:
        exclude:
          - "lib/theme/app_colors.dart"
          - "lib/**/design_tokens/**"
```

The existing boolean form should keep working unchanged:

```yaml
      avoid_hardcoded_color: false
```

Defaults stay as they are, so this is additive and no existing project changes behavior.

I am aware that no rule in `altive_lints_plugin` currently reads per-rule options, so this may be a larger change than it looks — it likely means introducing an options-parsing path for diagnostics in general, not just for this rule. If that is out of scope or not a direction you want to take, I would be glad to hear it; a narrower alternative is discussed below.

## Points for discussion

1. **Scope of the mechanism.** Per-rule `exclude` could be generally useful (`avoid_hardcoded_japanese` has had comparable requests in #47 / #65), so it may be worth designing it as a shared facility rather than as something specific to this rule.
2. **Narrower alternative.** If configuration is undesirable, adding a `ThemeExtension` subclass branch alongside `_isInsideColorScheme` would cover part of the problem without any new configuration surface. It would not help plain constant classes, though.
3. **`Colors.transparent`.** Not proposed for change here. #59 settled it as always allowed, and I am only noting that a project wanting to route even transparency through its own tokens cannot currently do so. I do not think that should block this issue.
4. **Interaction with suppressions.** If a path is excluded by configuration, existing `// ignore:` comments in that file become redundant; whether that should be surfaced as an "unnecessary ignore" hint is probably out of scope.

## Naming

If this is implemented as configuration on the existing rule, no new name is needed. If you would rather it be a separate, stricter rule, names consistent with the existing set would be:

- `avoid_hardcoded_color_outside_tokens`
- `prefer_color_token_reference`
- `avoid_color_literal_outside_definition`

My preference is to extend the existing rule, since the detection logic is identical and a second rule would mostly duplicate it.

I would be happy to open a PR implementing this if you think the direction is right. I would appreciate guidance first on whether per-rule options should be introduced as a shared facility or scoped to this rule alone.
